tests/starknet/scarb.rs

```diff
@@ -0,0 +1,43 @@
+use anyhow::Error;
+use scarb::{
+    core::{Config, PackageId, PackageName, SourceId, TargetKind},
+    ops::{self, CompileOpts, FeaturesOpts, FeaturesSelector},
+};
+use semver::Version;
+use std::{fs::canonicalize, path::PathBuf};
+
+#[allow(dead_code)]
+pub struct Compiler {
+    toml: PathBuf,
+    opts: CompileOpts,
+    packages: Vec<PackageId>,
+}
+
+#[allow(dead_code)]
+impl Compiler {
+    pub fn new(toml: &str) -> Result<Self, Error> {
+        let toml_absolute = canonicalize(PathBuf::from(toml))?;
+        let opts = CompileOpts {
```

Comment:
If `opts` and `packaged` are hard-coded and never changed, why keep them in `Compiler` as fields and not just use them in `compile`? Then whole `compile` seems to be a single function that takes `toml: &str` and returns `Result`.

Reply:
20 lines of code is not a hure function. Having a single function is better that having a pointless struct, don't you think?